src/semidiscretization_euler_gravity.jl

```diff
@@ -0,0 +1,304 @@
+  ParametersEulerGravity(background_density, gravitational_constant, cfl, n_iterations_max, timestep_gravity)
+end
+
+function Base.show(io::IO, parameters::ParametersEulerGravity)
+  print(io, "ParametersEulerGravity(")
+  print(io,   "background_density=", parameters.background_density)
+  print(io, ", gravitational_constant=", parameters.gravitational_constant)
+  print(io, ", cfl=", parameters.cfl)
+  print(io, ", n_iterations_max=", parameters.n_iterations_max)
+  print(io, ", timestep_gravity=", parameters.timestep_gravity)
+  print(io, ")")
+end
+function Base.show(io::IO, ::MIME"text/plain", parameters::ParametersEulerGravity)
+  println(io, "ParametersEulerGravity using")
+  println(io, "- background_density:     ", parameters.background_density)
+  println(io, "- gravitational_constant: ", parameters.gravitational_constant)
+  println(io, "- cfl (gravity):    ", parameters.cfl)
+  println(io, "- n_iterations_max: ", parameters.n_iterations_max)
+  print(io,   "- timestep_gravity: ", parameters.timestep_gravity)
+end
+
+
+"""
+    SemidiscretizationEulerGravity
+
+A struct containing everything needed to describe a spatial semidiscretization
+of a the compressible Euler equations with self-gravity, reformulating the
+Poisson equation for the gravitational potential as steady-state problem of
+the hyperblic diffusion equations.
+- Schlottke-Lakemper, Winters, Ranocha, Gassner (2020)
+  "A purely hyperbolic discontinuous Galerkin approach for self-gravitating gas dynamics"
+  [arXiv: 2008.10593](https://arXiv.org/abs/2008.10593)
+"""
+struct SemidiscretizationEulerGravity{SemiEuler, SemiGravity,
+                                      Parameters<:ParametersEulerGravity, Cache} <: AbstractSemidiscretization
+  semi_euler::SemiEuler
+  semi_gravity::SemiGravity
+  parameters::Parameters
+  performance_counter::PerformanceCounter
+  gravity_counter::PerformanceCounter
+  cache::Cache
+
+  function SemidiscretizationEulerGravity{SemiEuler, SemiGravity, Parameters, Cache}(
+      semi_euler::SemiEuler, semi_gravity::SemiGravity,
+      parameters::Parameters, cache::Cache) where {SemiEuler, SemiGravity,
+                                                   Parameters<:ParametersEulerGravity, Cache}
+    @assert ndims(semi_euler) == ndims(semi_gravity)
+    @assert typeof(semi_euler.mesh) == typeof(semi_gravity.mesh)
+    @assert polydeg(semi_euler.solver) == polydeg(semi_gravity.solver)
+
+    performance_counter = PerformanceCounter()
+    gravity_counter = PerformanceCounter()
+
+    new(semi_euler, semi_gravity, parameters, performance_counter, gravity_counter, cache)
+  end
+end
+
+function SemidiscretizationEulerGravity(semi_euler::SemiEuler, semi_gravity::SemiGravity, parameters) where
+    {Mesh, SemiEuler<:SemidiscretizationHyperbolic{Mesh, <:AbstractCompressibleEulerEquations},
+           SemiGravity<:SemidiscretizationHyperbolic{Mesh, <:AbstractHyperbolicDiffusionEquations}}
+
+  u_ode = compute_coefficients(0.0, semi_gravity)
```

Comment:
Since we use DIffEq to integrate the Euler part in time, intermediate caches for the time integration methods are stored there. We use some custom implementation of time integration methods to solve the hyperbolic diffusion equations to steady state. Since we don't use Diffeq for that right now, we have to store the caches necessary for the time integrator somewhere. Here, I chose to use the `cache` of the `SemidiscretizationEulerGravity` for that.
```julia
  u_ode = compute_coefficients(zero(real(semi_gravity)), semi_gravity)
  du_ode     = similar(u_ode)
  u_tmp1_ode = similar(u_ode)
  u_tmp2_ode = similar(u_ode)
  cache = (; u_ode, du_ode, u_tmp1_ode, u_tmp2_ode)
```

Reply:
OK, I think I understand it now... Thanks